.github/workflows/build.yml

```diff
@@ -33,6 +33,22 @@ jobs:
           # it relies on the existence of a go.sum file.
           cache: false
           go-version: "1.20"
+      - id: setup-go-critic
+        run: go install -v github.com/go-critic/go-critic/cmd/gocritic@latest
+      - id: setup staticheck
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - name: Run staticcheck
+        uses: dominikh/staticcheck-action@v1.3.0
+        with:
+          version: "2022.1.3"
+      - id: setup-go-sec
+        uses: actions/checkout@v3
+        env:
+          GO111MODULE: on
+      - name: Run Gosec Security Scanner
+        uses: securego/gosec@master
```

Comment:
- I may be wrong, but it seems that some (possibly all) of these tools are run via `pre-commit`.  What is the advantage of running them here as well?
- Shouldn't we be using the same version of `actions/checkout` throughout this entire file?
- What are these `actions/checkout` tasks doing, since they don't define a repository from which to checkout code?

Reply:
The skeleton golang build is missing these third-party packages since they are not included "out-of-the-box" with Go.  You are right, I will make all of the checkout versions the same.  actions/checkout clones the current repo if no repo is specified and for gosec and go-critic  we would want to execute this in the current build repo. Gosec just inspects the source for security issues. And go-critic is basically a linter.

Reply:
@jsf9k No, you're right. I get  you now. It's been awhile since I made this change and completely forgot

Reply:
@jasonodoom - Please allow the person who made the original comment to resolve the conversation.  That is the person in the best position to determine whether the issue has been correctly understood and completely addressed.

Reply:
- Unaddressed
- Addressed in commit 53f56b0
- Addressed in commit 097b573

Reply:
@jsf9k, no you're right. They shouldn't have been run here.

Reply:
I'm still confused.  You're adding `go-critic` and `staticcheck` to the GH Actions build workflow, but [it appears that those two tools are already being run via `pre-commit`](https://github.com/cisagov/skeleton-generic/blob/bd762fe4c4b41eca45fe8f0e3f3fa26bbfa0345c/.pre-commit-config.yaml#L67-L70) in the [`lint` job of the `build` workflow](https://github.com/cisagov/skeleton-generic/blob/bd762fe4c4b41eca45fe8f0e3f3fa26bbfa0345c/.github/workflows/build.yml#L100C1-L100C1).  What is the advantage to running them again outside of `pre-commit`?  Please point out what I am missing.

Reply:
@jsf9k Aaah, the [commands weren't found](https://github.com/cisagov/skeleton-golang-package/actions/runs/5490711848/jobs/10006481434) so this change was to make them avaialble. My understanding was that the pre-commit doesn't actually include those packages.

Reply:
OK, I understand now.